### ISSUE TITLE
Update build image to the latest `buildkit` change

### ISSUE THREAD

**Issue by potiuk** [apache/airflow#20946]
We have just merged the #20664 change that heavily simplified the way how our images are build and cached. We did not have any task implemented so far to cover those caching (I worked on this simplification in parallel)  so the changes should affect Breeze2 only very slightly.

The nice thing is that we have 700 (!) less lines of `bash` by just this change so the Breeze/CI Python rewrite project suddenly became way simpler :).

Those are the changes that I think affect current breeze2 work and should be implemented:

* no `--force-pull` flag - the caching mechanism of buildkit is so good that we do not have to force pull image any more
* no `-check-if-base-python-image-updated` flag - we do not keep base python image any more in our registry and we do not need to refresh the base image. Base image will be refreshed automatically as we also warn the users that they should always rebase to latest branch HEAD to rebuild the image
* no `push-image` command - we do not need to push images any more. instead we have `prepare-build-cache` command which builds images locally and pushes cache for them.
* no more complex "check if image needs pull" logic is needed any more.

**Comment by potiuk** [comment]
cc: @edithturn

**Comment by edithturn** [comment]
Got it @potiuk, do you want me to take this task? it will be great :)

**Comment by potiuk** [comment]
cc: @Bowrna

**Comment by Bowrna** [comment]
@potiuk do you think with buildkit changes, do we have to make changes in build-image in Breeze2? Could you explain to me more how this buildkit change will affect Breeze2?
https://github.com/apache/airflow/blob/ff4939f23ca11e77705fb0a54fe9d3b37375ec05/dev/breeze/src/airflow_breeze/ci/build_params.py#L33-L36

**Comment by potiuk** [comment]
Just a little. Actually if anything, The "buildkit" command made it the "old" breeze build much closer to what the "new" breeze did (i.e. simplified it).

What we really need to do is just syncing the parameters that are used to prepare the "docker build" command (see the description of the issue above):

* --force-pull  and --check-if-python-bae are gone
* build is always executed as one step for both PROD and CI image (previously PROD image was build in two steps) - so no change vs. what you've done already
* --all the code responsible for pulling images for cache is gone now (so you do not need to implement it).

So this one is really to make sure that we have the same parameters in build command than what we have in "old" breeze.

**Comment by potiuk** [comment]
One thing - but this should be a separate issue - is that we have this `prepare-build-cache` which slightly modifies build comman (adds --cache-to and starts builder before) - but this is something we should add after and the whole part (it is guarded by `PREPARE_BUILDX_CACHE` variable) about this can be skipped.

**Comment by Bowrna** [comment]
ok @potiuk I think I have to go and dig deeper to understand the impact of` buildkit` change. Could you tell me which command in breeze I can go through to better understand on the parts you have mentioned?

**Comment by potiuk** [comment]
Just
* build-image
* prepare-image-cache

That's all. Those are the only two impacted "really", we call "build-image" methods from other commands (for example when we run `./breeze` - build image is run when we determine if we should do it, but they use the same "_build_images.sh" libraries to call build image methods.)
